Fixes #3748 - Changed file names of EmailDashboardService according t…

ajomadlabs opened the pull request:
Fixes #3748 

Found that `EmailDashboardServices.js` was not following the guidelines as mentioned in #3748 .
Edited the file name of `EmailDashboardServices.js` and `EmailDashboardServicesSpec.js` to 
`EmailDashboardService.js` and `EmailDashboardServiceSpec.js` respectively.

ajomadlabs committed:
Fixes #3748 - Changed file names of EmailDashboardService according to the given guidelines

seanlip commented:
Hi @ajomadlabs, did you test this manually (e.g. by logging in as an admin and going to the email_dashboard page)?

ajomadlabs commented:
@seanlip I did manually run the test according to the guidelines given in 
`https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia`

seanlip commented:
I meant, setting up a dev server and going to the /email_dashboard page. Our test suite is not yet comprehensive, so it's important to test manually to ensure that nothing breaks.

ajomadlabs commented:
@seanlip I will check it now

seanlip commented:
Thanks! I'm also going to pass this review over to @prasanna08 (who wrote the email dashboard), so please correspond with him going forward.

ajomadlabs commented:
@prasanna08 I am getting a 404 error while going to `/email_dashboard`

seanlip commented:
Ah -- @ajomadlabs, please read https://github.com/oppia/oppia/wiki/Overview-of-the-Oppia-codebase and take a look at main.py for the exact URL.

ajomadlabs committed:
Merge branch 'develop' into angular-service-naming

Synced with upstream

ajomadlabs commented:
@seanlip @prasanna08 I had a look through the main.py and found that route was `/emaildashboard` but now I'm getting 
`
401 - Unauthorized
You can't get in there. Quick, go back before the teacher comes!`

ajomadlabs commented:
@seanlip Can you help me with this 401 error that is occuring

ajomadlabs committed:
Merge branch 'develop' into angular-service-naming

Synced with the latest commit

ajomadlabs commented:
@seanlip @prasanna08 When I tested manually I found that an error is being raised telling `'You do not have credentials to access email dashboard.'`. Can I know how I could access the emaildashboard locally

ajomadlabs committed:
Corrected the name of service in the html file

prasanna08 commented:
Hi @ajomadlabs to test the email dashboard on frontend you'll have to do a certain things.
1. Login as admin.
2. Go to `/admin` url. There go to config tab.
3. There you will find many different properties. Look for a property `Names of users allowed to send emails via the query interface.`.
4. Add your username there. The one that you used during login. Go to the bottom of the page and click `save`.
5. Once you see `Data Saved Successfully` in the bottom right corner of page, you're done. Now try to navigate to email dashboard link. It should be visible now.

ajomadlabs committed:
Merge branch 'develop' into angular-service-naming

Synced with the latest commit

ajomadlabs committed:
Changed the EmailDashboardDataService naming according to the given guidelines

ajomadlabs commented:
Workflow 

1. Went to `core/templates/dev/head/pages/email_dashboard`
2. Found that `EmailDashboardDataServices` was not following the naming guidelines
3. Opened the respective file `EmailDashboardDataServices`, checked whether it had just one service.
4. Renamed `EmailDashboardDataServices.js` to `EmailDashboardDataService.js`
5. Checked the `EmailDashboardDataServicesSpec.js`.
6. Renamed `EmailDashboardDataServicesSpec.js` to `EmailDashboardDataServiceSpec.js`
7. Opened `EmailDashboard.html` file and corrected the script src.
8. Script src was corrected with `EmailDashboardDataService.js`.

Tested Manually by checking if the link `/emaildashboard` is working.
All the test was passed successfully.

Screenshot
![emaildashboard](https://user-images.githubusercontent.com/21251118/29505723-80dafba6-8665-11e7-90ec-685d9b4d22ad.png)

@seanlip @prasanna08 Have tested manually. Everything works perfectly fine. Can you review it  and correct me if I am wrong

ajomadlabs commented:
@seanlip @prasanna08 I'm getting 
` Failed: unknown error: Element <button ng-if="!stateSolutionService.savedMemento" type="button" class="btn btn-default btn-lg oppia-add-hint-button protractor-test-oppia-add-solution-button" ng-click="openAddOrUpdateSolutionModal()">...</button> is not clickable at point (423, 876). Other element would receive the click: <div class="modal-backdrop fade leave leave-active" modal-animation-class="fade" modal-in-class="in" ng-style="{'z-index': 1040 + (index &amp;&amp; 1 || 0) + index*10}" modal-backdrop="modal-backdrop" modal-animation="true" style="z-index: 1040;"></div>`
Can you help me with this

ajomadlabs committed:
Change the file name

ajomadlabs commented:
@seanlip @prasanna08 I have solved and passed all checks . I request you to just review the changes

seanlip commented:
@ajomadlabs -- @prasanna08 is assigned to this PR, and should conduct this review. He's not online 24/7, so please give him 24 hours, though. Thanks!

If you like, you're welcome to start another PR in parallel. You don't have to wait for a review to complete before submitting another PR.

ajomadlabs commented:
@seanlip Thanks

seanlip commented on core/templates/dev/head/pages/email_dashboard/EmailDashboardDataService.js:
```diff
@@ -11,11 +11,10 @@
  * @fileoverview Services for oppia email dashboard page.
- */
-
+*/
```
This should be indented by 1 more space, as it was before.

ajomadlabs replied:
Sure will do it

seanlip commented on core/templates/dev/head/pages/email_dashboard/EmailDashboardDataService.js:
```diff
@@ -11,11 +11,10 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
```
Please preserve the newline that used to be below this one. In general, it's a good idea to check the "Files Changed" tab on GitHub to make sure that the diff is exactly what you want it to be :)

seanlip reviewed (changes requested):
Hi @ajomadlabs -- sorry, turns out @prasanna08 is having exams this week! I've taken a look and it looks good apart from a couple of small things. Could you please fix them? Then we can merge this.

Thanks!

ajomadlabs committed:
Updated EmailDashboardDataService and have followed the coding guidelines.

ajomadlabs commented:
@seanlip @prasanna08 Have modified according to the requested changes.

ajomadlabs reviewed (commented):
@seanlip I have fixed and followed according to the coding standards.

seanlip commented on core/templates/dev/head/pages/email_dashboard/EmailDashboardDataService.js:
```diff
@@ -15,7 +15,7 @@
  * @fileoverview Services for oppia email dashboard page.
  */
-
+ 
```
Same comment as before -- please check the diff, make sure there are no unnecessary changes.

Otherwise, looks good!

ajomadlabs committed:
Updated EmailDashboard according to given changes

seanlip reviewed (approved):
LGTM. Thanks @ajomadlabs!
